ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/auth/PermissionChecker.java

```diff
@@ -94,15 +94,10 @@ private boolean isChangingRestrictedSettings(RoleOptions options)
 
     private boolean hasPermissionToAlterRole(AuthenticatedUser performer, RoleResource roleResource)
```

Comment:
Since we are passing around perfomer/role/roleResource everywhere, should we break it out into it's own context, i.e:

```
private class PermissionContext
{
 final AuthenticatedUser performer;
 final RoleResource role;
 final RoleOptions roleOptions;
}
```

and add all the methods without the arguments to it. By that we could do i.e:
`.map(this::getPermissions)`
in the stream below.

Not sure if there is any actual benefit to this though.

Reply:
Interesting idea. I propose to evaluate this in a separate ticket.